### ISSUE TITLE
Does nd4j support to apply a user defined function to each element value of NDArray?

### ISSUE THREAD

**Issue by wenbostar** [deeplearning4j/nd4j#1791]
#### Issue Description

Does nd4j support to apply a user defined function to each element value of NDArray?

Best regards!
Bo

**Comment by saudet** [comment]
User defined as in Java? Sounds like a duplicate of issue deeplearning4j/nd4s#91.

**Comment by wenbostar** [comment]
@saudet , yes, apply a function defined by user to each element of an NDArray object.

**Comment by wenbostar** [comment]
@saudet ，it does not support this function currently？

**Comment by saudet** [comment]
No it doesn't.

**Comment by wenbostar** [comment]
@saudet will it support in the future?

**Comment by saudet** [comment]
Eventually, we would like to, but it's unclear how to go about it since we also need to care about CUDA and other such native computing platforms. If you have ideas though, by all means, let's try them out!

**Comment by raver119** [comment]
Well, nd4j support somewhat like that, but:
1) That function will be applied sequentially, not concurrently.
2) It will be applied on java side, not in native.

That called BooleanIndexing

**Comment by huitseeker** [comment]
Note we also have these functions built-in:

https://github.com/deeplearning4j/dl4j-examples/blob/master/nd4j-examples/src/main/java/org/nd4j/examples/Nd4jEx9_Functions.java

**Comment by raver119** [comment]
Just update: we're going to introduce lambda support on java side soon.
